src/analysis/analysis-factory.js

```diff
@@ -51,9 +51,14 @@ AnalysisFactory.prototype.analyse = function (analysisDefinition) {
 AnalysisFactory.prototype._getAnalysisAttributesFromAnalysisDefinition = function (analysisDefinition) {
   var analysisType = analysisDefinition.type;
   var sourceNamesForAnalysisType = this._camshaftReference.getSourceNamesForAnalysisType(analysisType);
+  var optionalSourceNamesForAnalysisType = this._camshaftReference.getOptionalSourceNamesForAnalysisType(analysisType);
   var sourceNodes = {};
   _.each(sourceNamesForAnalysisType, function (sourceName) {
-    sourceNodes[sourceName] = this.analyse(analysisDefinition.params[sourceName]);
+    var isOptional = _.indexOf(optionalSourceNamesForAnalysisType, sourceName) > -1;
+    var sourceParams = analysisDefinition.params[sourceName];
+    if (sourceParams || !isOptional) {
```

Comment:
It'll break just as it breaks right now before this PR. If the param is required but it has no value, it should break.

In practice this doesn't happen because we didn't have analysis with optional node as a param (until now).